### PR TITLE
Custom tags

### PR DESCRIPTION
**Pull request by DominikaK**
(2.1)

Based on https://github.com/ezsystems/ezplatform-admin-ui/pull/320#issuecomment-360469462 and https://github.com/alongosz/ezplatform/commit/e98f3fd117adde76a0f2b221eef22ce6f3dbb2e1

### CODE REVIEW

**Comment by DominikaK** [comment]
ping @alongosz @dew326

File: docs/guide/extending_ez_platform_ui.md [hunk 1]
```diff
@@ -661,6 +661,77 @@ Optionally, Sub-items module can take a following list of props:
...
+                        type: 'boolean'
+                        default_value: false
+                    align:
+                        type: 'choice'
+                        required: false
+                        default_value: 'left'
+                        choices: ['left', 'center', 'right']
+```
+
+Each custom tag can have any number of attributes. Attributes can be of one of four types:
```
Comment (MagdalenaZuba) [inline docs/guide/extending_ez_platform_ui.md]: (Supported) attributes types:

File: docs/guide/extending_ez_platform_ui.md [hunk 2]
```diff
@@ -661,6 +661,77 @@ Optionally, Sub-items module can take a following list of props:
...
+                        default_value: false
+                    align:
+                        type: 'choice'
+                        required: false
+                        default_value: 'left'
+                        choices: ['left', 'center', 'right']
+```
+
+Each custom tag can have any number of attributes. Supported attribute types are:
+`string`, `number`, `boolean` and `choice`.
```
Comment (alongosz) [inline docs/guide/extending_ez_platform_ui.md]: `choice` attribute requires list of choices provided by `choices` key. (_or something like that_ :grinning: )

Also worth to mention: if `required` key is omitted, it is assumed to be `false`

File: docs/guide/extending_ez_platform_ui.md [hunk 3]
```diff
@@ -661,6 +661,77 @@ Optionally, Sub-items module can take a following list of props:
...
+The configuration requires a Twig template for the custom tag:
+
+``` twig
+<div{% if params.align is defined %} style="text-align: {{ params.align }};"{% endif %}>
+    <iframe type="text/html" width="{{ params.width }}" height="{{ params.height }}"
+        src="{{ params.video_url|replace({'https://youtu.be/' : 'https://www.youtube.com/embed/'}) }}?autoplay={{ params.autoplay == 'true' ? 1 : 0 }}"
+        frameborder="0"></iframe>
+</div>
+```
+
```
Comment (alongosz) [inline docs/guide/extending_ez_platform_ui.md]: Remember that if attribute is not required, then you need to check in your Twig template if it is defined:
```twig
{% if params.your_attribute is defined %} something which uses {{ params.your_attribute }}. {% endif %}
```

File: docs/guide/extending_ez_platform_ui.md [hunk 4]
```diff
@@ -661,6 +661,77 @@ Optionally, Sub-items module can take a following list of props:
...
+    system:
+        default:
+            fieldtypes:
+                ezrichtext:
+                    custom_tags: [ezyoutube]
+
+    ezrichtext:
+        custom_tags:
+            ezyoutube:
+                template: 'AppBundle:field_type/ezrichtext/custom_tag:ezyoutube.html.twig'
```
Comment (alongosz) [inline docs/guide/extending_ez_platform_ui.md]: maybe let's add above:
```twig
# A template used for frontend rendering of Custom Tag
```

File: docs/guide/extending_ez_platform_ui.md [hunk 5]
```diff
@@ -661,6 +661,77 @@ Optionally, Sub-items module can take a following list of props:
...
+                icon: '/assets/field_type/ezrichtext/custom_tag/icon/youtube-color.svg#youtube-color'
```
Comment (alongosz) [inline docs/guide/extending_ez_platform_ui.md]: let's also add above this comment:
```twig
# An icon representing Custom Tag displayed in the Online Editor toolbar.
```

**Pull request merged**